src/Component/AbstractGrid.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid;
+
+use Sylius\Component\Grid\Builder\GridBuilder;
+use Sylius\Component\Grid\Builder\GridBuilderInterface;
+use Sylius\Component\Grid\Definition\Grid;
+
+abstract class AbstractGrid implements GridInterface
+{
+    public function getDefinition(): Grid
+    {
+        $gridBuilder = GridBuilder::create(static::getName(), static::getResourceClass());
+
+        $this->buildGrid($gridBuilder);
+
+        return $gridBuilder->getDefinition();
+    }
+
+    protected function buildGrid(GridBuilderInterface $gridBuilder): void
```

Comment:
Why isn't this function abstract. A grid without any definition doesn't make sense to me.

Reply:
Yes I think you're right. I was inspired by the Symfony\Component\Form\AbstractType and it does not have any abstract methods.
Maybe I could make it public and add it on the grid interface and remove id from AbstractGrid.
@pamil WDT ?

Reply:
@mamazu I added the buildGrid method on the grid interface 🎉

Reply:
Great, then this should be resolved.